dlq_processor/cloudwatchlogs_lambda.js

```diff
@@ -139,30 +44,33 @@ function getConfig(env) {
 }
 
 
-exports.processLogs = function (env, context, eventAwslogsData) {
+exports.processLogs = function (env, eventAwslogsData, errorHandler) {
 
     var config = getConfig(env);
+    var SumoLogsClientObj = new SumoLogsClient(config);
 
     // Used to hold chunks of messages to post to SumoLogic
     var messageList = {};
 
     // Validate URL has been set
     var urlObject = url.parse(config.SumoURL);
-    if (urlObject.protocol != 'https:' || urlObject.host === null || urlObject.path === null) {
-        throw 'Invalid SUMO_ENDPOINT environment variable: ' + config.SumoURL;
+    if (urlObject.protocol !== 'https:' || urlObject.host === null || urlObject.path === null) {
+        errorHandler('Invalid SUMO_ENDPOINT environment variable: ' + config.SumoURL, 'Error in SumoURL');
```

Comment:
Shouldn't the first param to errorHandler be an Error object?

Reply:
though it is not wrong but will change it to Error object because the later gives stacktrace as well.